src/NexusMods.CLI/CliGuidedInstaller.cs

```diff
@@ -0,0 +1,241 @@
+using System.Diagnostics;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.CLI;
+using NexusMods.Abstractions.CLI.DataOutputs;
+using NexusMods.Common;
+using NexusMods.Common.GuidedInstaller;
+
+namespace NexusMods.CLI;
+
+/// <summary>
+/// Implementation of an option selector for the CLI.
+/// </summary>
+[UsedImplicitly]
+public class CliGuidedInstaller : IGuidedInstaller
+{
+    private const string CancelInput = "x";
+    private const string BackInput = "b";
+    private const string NextInput = "n";
+
+    private static readonly string[] TableOfGroupsHeaders = { "Key", "Group" };
+    private static readonly object[] TableOfGroupsFooterNextGroup = { NextInput, "Next Step" };
+    private static readonly object[] TableOfGroupsFooterFinish = { NextInput, "Finish Installation" };
+    private static readonly object[] TableOfGroupsFooterGoBack = { BackInput, "Previous Step" };
+    private static readonly object[] TableOfGroupsFooterCancel = { CancelInput, "Cancel Installation" };
+
+    private static readonly string[] TableOfOptionsHeaders = { "Key", "State", "Name", "Description" };
+    private static readonly object[] TableOfOptionsFooterBackToGroupSelection = { BackInput, "", "Back", "Back to the group selection" };
+    private static readonly object[] TableOfOptionsFooterCancel = { CancelInput, "", "Cancel", "Cancel the installation" };
+
+    private readonly ILogger<CliGuidedInstaller> _logger;
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    public CliGuidedInstaller(ILogger<CliGuidedInstaller> logger)
+    {
+        _logger = logger;
+    }
+
+    /// <summary>
+    /// The renderer to use for rendering the options.
+    /// </summary>
+    public IRenderer Renderer { get; set; } = null!;
+
+    /// <summary>
+    /// If true, all option selection will select the first choice.
+    /// </summary>
+    public bool SkipAll { get; set; }
+
+    /// <inheritdoc/>
+    public void SetupInstaller(string name) { }
+
+    /// <inheritdoc/>
+    public void CleanupInstaller() { }
+
+    /// <inheritdoc />
+    public Task<UserChoice> RequestUserChoice(GuidedInstallationStep installationStep, CancellationToken cancellationToken)
+    {
+        OptionGroup? currentGroup = null;
+
+        var selectedOptions = installationStep.Groups
+            .SelectMany(group => group.Options
+                .Where(option => option.Type == OptionType.PreSelected)
+                .Select(option => new SelectedOption(group.Id, option.Id))
+            ).ToList();
+
+        while (true)
+        {
+            if (currentGroup is null)
+            {
+                // if the installation step has multiple groups
+                // the user has to select which group they want to use
+
+                RenderTableOfGroups(installationStep);
```

Comment:
So this is always null the first time around. I'm guessing that even if there is just one group, you still show the table to let the user select it first?
The comment might be confusing, but perhaps only because I know how it worked before and that influenced me.